### PR TITLE
Havardpe/overload

### PR DESCRIPTION
**Pull request by havardpe**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@vekterli please review
@arnej27959 @geirst FYI

### CODE REVIEW

File: vespalib/src/vespa/vespalib/util/visit_ranges.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+#include <functional>
+
+namespace vespalib {
+
+struct visit_ranges_either {};
+struct visit_ranges_first : visit_ranges_either {};
+struct visit_ranges_second : visit_ranges_either {};
+struct visit_ranges_both {};
+
```
Comment (arnej27959) [inline vespalib/src/vespa/vespalib/util/visit_ranges.h]: visit_ranges should have a documentation comment explaining its requirements.

**Review by arnej27959 — APPROVED** [review]
looks good

**Review by vekterli — APPROVED** [review]
:+1:

**Review by arnej27959 — APPROVED** [review]
👍

**Pull request merged**